contracts/test/plugin/plugin-uups-upgradeable.ts

```diff
@@ -1,50 +1,84 @@
 import {
+  DAOMock,
+  DAOMock__factory,
   IERC1822ProxiableUpgradeable__factory,
   IPlugin__factory,
-  DAOMock__factory,
   IProtocolVersion__factory,
   PluginUUPSUpgradeableMockBuild1,
   PluginUUPSUpgradeableMockBuild1__factory,
   PluginUUPSUpgradeableMockBuild2__factory,
+  PluginUUPSUpgradeableMockBad__factory,
+  ProxyFactory__factory,
 } from '../../typechain';
+import {
+  ProxyCreatedEvent,
+  ProxyFactory,
+} from '../../typechain/src/utils/deployment/ProxyFactory';
 import {erc165ComplianceTests, getOzInitializedSlotValue} from '../helpers';
 import {osxCommonsContractsVersion} from '../utils/versioning/protocol-version';
 import {
+  findEvent,
   getInterfaceId,
   PLUGIN_UUPS_UPGRADEABLE_PERMISSIONS,
   PluginType,
+  PROXY_FACTORY_EVENTS,
 } from '@aragon/osx-commons-sdk';
+import {loadFixture} from '@nomicfoundation/hardhat-network-helpers';
 import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/signers';
 import {expect} from 'chai';
-import {ethers, upgrades} from 'hardhat';
+import {ethers} from 'hardhat';
+
+type FixtureResult = {
+  deployer: SignerWithAddress;
+  implementation: PluginUUPSUpgradeableMockBuild1;
+  proxyFactory: ProxyFactory;
+  daoMock: DAOMock;
+  initCalldata: string;
+  Build1Factory: PluginUUPSUpgradeableMockBuild1__factory;
+  Build2Factory: PluginUUPSUpgradeableMockBuild2__factory;
+};
+
+async function fixture(): Promise<FixtureResult> {
+  const [deployer] = await ethers.getSigners();
+
+  const Build1Factory = new PluginUUPSUpgradeableMockBuild1__factory(deployer);
+  const Build2Factory = new PluginUUPSUpgradeableMockBuild2__factory(deployer);
+  const daoMock = await new DAOMock__factory(deployer).deploy();
+
+  const implementation = await Build1Factory.deploy();
+
+  const proxyFactory = await new ProxyFactory__factory(deployer).deploy(
+    implementation.address
+  );
+
+  const initCalldata = implementation.interface.encodeFunctionData(
+    'initialize',
+    [daoMock.address]
+  );
+
+  return {
+    deployer,
+    implementation,
+    proxyFactory,
+    initCalldata,
+    daoMock,
+    Build1Factory,
+    Build2Factory,
+  };
+}
```

Comment:
move it either to the bottom or in all the other files to the top so that the fixture is always in the same location

Reply:
Done in 3d156e7e6accd773f4bc1a0af210556f083dc251